File: pkg/dart2cpp/sample/dart/test_arrow_operator.dart
void main() {
  // 测试基础类型的 -> 运算符
  var x = 10;
  var y = 20;
  
  // 测试 Int 类型
  print('x + y = ${x + y}');
  print('x.abs() = ${x.abs()}');
  
  // 测试 Double 类型
  var d1 = 3.14;
  var d2 = 2.71;
  print('d1 + d2 = ${d1 + d2}');
  print('d1.abs() = ${d1.abs()}');
  
  // 测试 Bool 类型
  var b1 = true;
  var b2 = false;
  print('b1 && b2 = ${b1 && b2}');
  print('b1.toString() = ${b1.toString()}');
}

File: pkg/dart2cpp/sample/cpp_generated/test_arrow_operator.cpp
#include "dart2cpp.h"

// 工具宏定义

// ============================================================================
// 主函数
// ============================================================================

int main() {
  try {
    auto x = dart_int(10);
auto y = dart_int(20);
dart_print(dart_string("x + y = ") + (x->operator_add(y)).toString());
dart_print(dart_string("x.abs() = ") + (x->abs()).toString());
auto d1 = dart_double(3.14);
auto d2 = dart_double(2.71);
dart_print(dart_string("d1 + d2 = ") + (d1->operator_add(d2)).toString());
dart_print(dart_string("d1.abs() = ") + (d1->abs()).toString());
auto b1 = dart_bool(true);
auto b2 = dart_bool(false);
dart_print(dart_string("b1 && b2 = ") + (b1 && b2).toString());
dart_print(dart_string("b1.toString() = ") + (b1->toString()).toString());
    return 0;
  } catch (const std::exception& e) {
    std::cerr << "Error: " << e.what() << std::endl;
    return 1;
  }
}
